contrib/platform/src/com/sun/jna/platform/win32/Guid.java

```diff
@@ -466,16 +468,14 @@ public CLSID(GUID guid) {
 
     /**
      * The Class REFIID.
-     *
-     * @author Tobias Wolf, wolf.tobias@gmx.net
      */
-    public class REFIID extends IID {
+    public class REFIID extends ByReference {
```

Comment:
ok, I misread this.  You should implement `REFIID` as a `PointerType` (I though you were using `Structure.ByReference` rather than `com.sun.jna.ptr.ByReference`).  `com.sun.jna.ptr.ByReference` should generally only be used when your called function expects a pointer to something else in order to "return" a value via parameter.

If you change `REFIID` to `PointerType` your code should still work and will not be ambiguous or misleading.